core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1653,7 +1660,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
   val metadataSnapshotMaxNewRecordBytes = getLong(KafkaConfig.MetadataSnapshotMaxNewRecordBytesProp)
 
   /************* Authorizer Configuration ***********/
-  val authorizer: Option[Authorizer] = {
+  val createNewAuthorizer: Option[Authorizer] = {
```

Comment:
This should be `def`. Otherwise, when running with `process.roles=broker,controller`, we will use the same `Authorizer` instance, which leads to duplicate errors.

Reply:
ugh. This explains some of the bugs we've seen, definitely. Yes, I'll fix that here...